src/utils/application.data.ts

```diff
@@ -10,41 +10,75 @@ import {
   OwnerOfLandOtherEntityJurisdictionsNoc,
   OwnerOfLandPersonJurisdictionsNoc,
   TrustControlNoc,
-  TrusteesNoc
+  TrusteesNoc,
+  OverseasEntityKey,
+  Transactionkey,
 } from '../model/data.types.model';
+
 import {
   ApplicationData,
   APPLICATION_DATA_KEY,
   ApplicationDataType,
   ApplicationDataArrayType
 } from "../model";
+
 import { BeneficialOwnerGov, BeneficialOwnerGovKey } from '../model/beneficial.owner.gov.model';
 import { BeneficialOwnerIndividual, BeneficialOwnerIndividualKey } from '../model/beneficial.owner.individual.model';
 import { BeneficialOwnerOtherKey } from '../model/beneficial.owner.other.model';
 import { ManagingOfficerCorporate, ManagingOfficerCorporateKey } from '../model/managing.officer.corporate.model';
 import { ManagingOfficerIndividual, ManagingOfficerKey } from '../model/managing.officer.model';
+
 import {
   PARAM_BENEFICIAL_OWNER_GOV,
   PARAM_BENEFICIAL_OWNER_INDIVIDUAL,
   PARAM_BENEFICIAL_OWNER_OTHER,
   PARAM_MANAGING_OFFICER_CORPORATE,
   PARAM_MANAGING_OFFICER_INDIVIDUAL,
-  FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC
+  FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC,
+  FEATURE_FLAG_ENABLE_REDIS_REMOVAL,
+  ROUTE_PARAM_TRANSACTION_ID,
+  ROUTE_PARAM_SUBMISSION_ID
 } from '../config';
+
 import { BeneficialOwnerCorporate } from '@companieshouse/api-sdk-node/dist/services/overseas-entities';
 import { Remove } from 'model/remove.type.model';
 import { isActiveFeature } from "./feature.flag";
 import { isNoChangeJourney } from "./update/no.change.journey";
+import { getOverseasEntity } from "../service/overseas.entities.service";
+
+export const getApplicationData = async (session: Session | undefined, req?: Request): Promise<ApplicationData> => {
```

Comment:
See earlier comment regarding this function signature change. Would an overloaded function with different params be better perhaps?

Reply:
I think that the first parameter will be deleted once finished so that only req stays

Reply:
Yes, have updated this now to use a single parameter as per your previous comment.